python/ray/_private/worker.py

```diff
@@ -2837,30 +2837,62 @@ def kill(actor: "ray.actor.ActorHandle", *, no_restart: bool = True):
 
 @PublicAPI
 @client_mode_hook
-def cancel(object_ref: "ray.ObjectRef", *, force: bool = False, recursive: bool = True):
-    """Cancels a task according to the following conditions.
-
-    If the specified task is pending execution, it will not be executed. If
-    the task is currently executing, the behavior depends on the ``force``
-    flag. When ``force=False``, a KeyboardInterrupt will be raised in Python
-    and when ``force=True``, the executing task will immediately exit.
-    If the task is already finished, nothing will happen.
-
-    Only non-actor tasks can be canceled. Canceled tasks will not be
-    retried (max_retries will not be respected).
-
-    Calling ray.get on a canceled task will raise a TaskCancelledError or a
-    WorkerCrashedError if ``force=True``.
+def cancel(
+    object_ref: "ray.ObjectRef", *, force: bool = False, recursive: bool = True
+) -> None:
+    """Cancels a task.
+
+    Cancel API has a different behavior depending on if it is a remote function
+    (Task) or a remote Actor method (Actor Task).
+
+    Task:
+        If the specified Task is pending execution, it is cancelled and not
+        executed. If the Task is currently executing, the behavior depends
+        on the `force flag. When `force=False, a KeyboardInterrupt is
+        raised in Python and when `force=True, the executing Task
+        immediately exits. If the Task is already finished, nothing happens.
+
+        Cancelled Tasks aren't retried. `max_task_retries` aren't respected.
+
+        Calling ray.get on a cancelled Task raises a TaskCancelledError if
```

Comment:
These different behaviors are strange, but I assume we can't really change it any more without API changes, right (i.e. this is already the behavior or Ray 2.6)?

Reply:
Oh actually I fixed this in the master because ray.cancel never specifies this behavior in the docstring. If you read the docstring, it says

```
    Calling ray.get on a canceled task will raise a TaskCancelledError or a
    WorkerCrashedError if ``force=True``.
```

So I considered as a bug fix, not API change. https://github.com/ray-project/ray/pull/38973. (I also brought this up to api review channel, https://anyscaleteam.slack.com/archives/CTSRWH3DJ/p1692224223820259)

Reply:
We then should update the doc to reflect the bug fix?

TaskCancelledError if the Task has been scheduled and interrupted,